eta/core/datasets.py

```diff
@@ -1159,11 +1186,37 @@ def __init__(self, video_path, labels_path, start_frame=1,
                                                   self.duration)
         self.labels_cls = etav.VideoLabels
 
-    def _set_video_metadata(self):
-        video_metadata = etav.VideoMetadata.build_for(self.data_path)
-        self.total_frame_count = video_metadata.total_frame_count
-        self.duration = video_metadata.duration
-        self.end_frame = video_metadata.total_frame_count
+    def _init_from_video_metadata(self):
+        self.total_frame_count = self.metadata.total_frame_count
+        self.duration = self.metadata.duration
+        self.end_frame = self.metadata.total_frame_count
+
+    @property
+    def metadata(self):
+        if not self._metadata:
+            self._metadata = etav.VideoMetadata.build_for(self.data_path)
```

Comment:
We avoided loading the metadata file directly elsewhere, will this require the file, or can this be constructed optionally from a provided dictionary of metadata fields?

Reply:
This can be avoided. I'll update this.